src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -568,28 +590,32 @@ namespace OpenMS
           }
           pos = (pos + ion_offset) / charge;
 
-          Peak1D p;
-          p.setMZ(pos);
-          p.setIntensity(intensity);
-          spectrum.push_back(p);
-          if (add_metainfo_)
-          {
-            String ion_name = String(Residue::residueTypeToIonLetter(res_type)) + String(i + 1) + String((Size)abs(charge), '+');
-            ion_names.push_back(ion_name);
-            charges.push_back(charge);
-          }
-
           if (add_losses_ && !add_isotopes_)
           {
             if (peptide[i].hasNeutralLoss())
             {
               for (const auto& formula : peptide[i].getLossFormulas()) fx_losses.insert(formula);
             }
             addLosses_faster_(spectrum, mono_weight + ion_offset, fx_losses,
-                i + 1, ion_names, charges, intensity * rel_loss_intensity_,
-                res_type, add_metainfo_, charge);
+                              i + 1, ion_names, charges, intensity * rel_loss_intensity_,
+                              res_type, add_metainfo_, charge);
+            chunks.emplace_back(spectrum.size(), false);
+          }
+
+          Peak1D p;
+          p.setMZ(pos);
+          p.setIntensity(intensity);
+          spectrum.push_back(p);
```

Comment:
or write a Ctor for Peak1D with `Peak1D(mz, intensity)`; then you can just `spectrum.emplace_back(pos, intensity);`

Reply:
The emplace_back version sounds good

Reply:
The constructor already exists if I'm not mistaken:
`cpp
inline Peak1D(PositionType a, IntensityType b) : position_(a), intensity_(b) {}`